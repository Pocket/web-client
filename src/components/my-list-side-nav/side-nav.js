import { css } from 'linaria'

export const sideNav = css`
  /* position: fixed;
  height: 100%;
  overflow: hidden; */
`

export const sideNavHeader = css`
  font-family: var(--fontSansSerif);
  font-size: var(--fontSize100);
  font-weight: 500;
  line-height: 0.8;
  padding: 20px 20px 10px;
  margin: 5px 0;
  color: var(--color-grey65); /* ! Don't use colors direct */
`

export const sideNavItem = css`
  display: flex;
  align-items: center;
  align-content: center;
  width: 100%;
  padding: 0 20px;
  font-size: var(--fontSize100);
  font-weight: 400;
  line-height: 16px;
  height: 40px;
  margin-bottom: 5px;
  border-radius: 4px;
  text-decoration: none;
  color: var(--color-textPrimary);
  background-color: transparent;

  &:hover {
    color: var(--color-actionPrimary);
    background-color: transparent;
  }
  &:active,
  &:focus {
    transition: none;
    color: var(--color-actionPrimary);
    outline: 1px auto var(--color-navCurrentTab);
  }
  &.active {
    color: var(--color-actionPrimaryHover);
    background-color: var(--color-navCurrentTab);
  }
`
