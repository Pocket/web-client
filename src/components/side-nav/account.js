import { sideNavHeader } from './side-nav'

export function AccountSideNav() {
  return (
    <>
      <div data-cy='side-nav-account' className={sideNavHeader}></div>
    </>
  )
}
