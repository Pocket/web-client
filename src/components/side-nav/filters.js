import Link from 'next/link'
import { Trans } from 'next-i18next'
import { css } from 'linaria'

import { FavoriteIcon } from '@pocket/web-ui'
import { HighlightIcon } from '@pocket/web-ui'
import { TagIcon } from '@pocket/web-ui'
import { ArticleIcon } from '@pocket/web-ui'
import { ArchiveIcon } from '@pocket/web-ui'
import { VideoIcon } from '@pocket/web-ui'

import { sideNavHeader } from './side-nav'

export function FiltersSideNav({
  showLab,
  subActive,
  pinnedTags
}) {
  return (
    <>
      <div className={sideNavHeader}>
        <Trans i18nKey="nav:filters">Filters</Trans>
      </div>

      {showLab ? (
        <Link href="/my-list/archive">
          <button className={subActive('archive')}>
            <ArchiveIcon className="side-nav-icon" />{' '}
            <Trans i18nKey="nav:archive">Archive</Trans>
          </button>
        </Link>
      ) : null}

      <Link href="/my-list/favorites">
        <button className={subActive('favorites')}>
          <FavoriteIcon className="side-nav-icon" />{' '}
          <Trans i18nKey="nav:favorites">Favorites</Trans>
        </button>
      </Link>

      <Link href="/my-list/highlights">
        <button className={subActive('highlights')}>
          <HighlightIcon className="side-nav-icon" />{' '}
          <Trans i18nKey="nav:highlights">Highlights</Trans>
        </button>
      </Link>

      <Link href="/my-list/articles">
        <button className={subActive('articles')}>
          <ArticleIcon className="side-nav-icon" />{' '}
          <Trans i18nKey="nav:articles">Articles</Trans>
        </button>
      </Link>

      <Link href="/my-list/videos">
        <button className={subActive('videos')}>
          <VideoIcon className="side-nav-icon" />{' '}
          <Trans i18nKey="nav:videos">Videos</Trans>
        </button>
      </Link>
      <div className={sideNavHeader}>Tags</div>
      <Link href="/my-list/tags">
        <button className={subActive('tag')}>
          <TagIcon className="side-nav-icon" />{' '}
          <Trans i18nKey="nav:all-tags">All Tags</Trans>
        </button>
      </Link>
      {pinnedTags.length
        ? pinnedTags.map((tag) => (
            <Link href={`/my-list/tags/${tag}`} key={tag}>
              <button className={subActive(tag, true)}>{tag}</button>
            </Link>
          ))
        : null}
    </>
  )
}
