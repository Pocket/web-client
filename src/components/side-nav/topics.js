import Link from 'next/link'
import { Trans } from 'next-i18next'
import { css } from 'linaria'

import { FavoriteIcon } from '@pocket/web-ui'
import { HighlightIcon } from '@pocket/web-ui'
import { TagIcon } from '@pocket/web-ui'
import { ArticleIcon } from '@pocket/web-ui'
import { ArchiveIcon } from '@pocket/web-ui'
import { VideoIcon } from '@pocket/web-ui'

import { sideNavHeader } from './side-nav'

export function TopicsSideNav({
  subActive,
  pinnedTopics,
  showVoltron
}) {
  return (
    <>
      <div className={sideNavHeader}>
        <Trans i18nKey="nav:topics">Saved Topics</Trans>
      </div>

      {pinnedTopics.length
        ? pinnedTopics.map((topic) => (
            <a href={`https://getpocket.com/explore/${topic.topic_slug}?src=sidebar`} key={topic.topic_slug}>
              <button className={subActive(topic)}>{topic.display_name}</button>
            </a>
          ))
        : null}
    </>
  )
}
