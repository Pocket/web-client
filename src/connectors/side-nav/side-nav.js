import { SideNav as SideNavComponent } from 'components/side-nav/side-nav'
import { useSelector } from 'react-redux'
export function SideNav({ subset, isLoggedIn, tag }) {
  const pinnedTags = useSelector((state) => state.userTags.pinnedTags)
  const appMode = useSelector((state) => state?.app?.mode)
  const isDisabled = appMode === 'bulk'
  return (
    <SideNavComponent
      isDisabled={isDisabled}
      subset={subset}
      isLoggedIn={isLoggedIn}
      pinnedTags={pinnedTags}
      tag={tag}
    />
  )
}
