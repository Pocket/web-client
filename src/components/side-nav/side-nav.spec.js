import { wrappedRender } from 'test-utils'
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils'
import '@testing-library/jest-dom/extend-expect'

import { SideNav } from './side-nav'

describe('SideNav', () => {
  const handleClick = () => {}

  const baseProps = {
    isDisabled: false,
    subset: 'tag',
    isLoggedIn: true,
    pinned: [
      'home',
      'space'
    ],
    tag: '_untagged_',
    newSaveCount: 0,
    flagsReady: true,
    trackMenuClick: handleClick
  }

  beforeAll(() => mockAllIsIntersecting())

  it('renders the My List side nav and shows tags', () => {
    const { queryByCy, queryAllByCy } = wrappedRender(<SideNav type='my-list' {...baseProps} />)

    // Main section
    expect(queryByCy('side-nav-home')).toBeInTheDocument()
    expect(queryByCy('side-nav-mylist')).toBeInTheDocument()
    expect(queryByCy('side-nav-discover')).toBeInTheDocument()
    expect(queryByCy('side-nav-collections')).toBeInTheDocument()

    // Filters section
    expect(queryByCy('side-nav-archive')).toBeInTheDocument()
    expect(queryByCy('side-nav-favorites')).toBeInTheDocument()
    expect(queryByCy('side-nav-highlights')).toBeInTheDocument()
    expect(queryByCy('side-nav-articles')).toBeInTheDocument()
    expect(queryByCy('side-nav-videos')).toBeInTheDocument()

    // Tags section
    expect(queryByCy('side-nav-all-tags')).toBeInTheDocument()
    expect(queryAllByCy(/side-nav-tags-(.+)/)).toHaveLength(2)

    // Snapshot
    expect(queryByCy('side-nav')).toMatchSnapshot()
  })

  it('renders the Account side nav', () => {
    const { queryByCy } = wrappedRender(<SideNav type='account' />)

    // Main section
    expect(queryByCy('side-nav-home')).toBeInTheDocument()
    expect(queryByCy('side-nav-mylist')).toBeInTheDocument()
    expect(queryByCy('side-nav-discover')).toBeInTheDocument()
    expect(queryByCy('side-nav-collections')).toBeInTheDocument()

    // Filters section
    expect(queryByCy('side-nav-archive')).not.toBeInTheDocument()
    expect(queryByCy('side-nav-favorites')).not.toBeInTheDocument()
    expect(queryByCy('side-nav-highlights')).not.toBeInTheDocument()
    expect(queryByCy('side-nav-articles')).not.toBeInTheDocument()
    expect(queryByCy('side-nav-videos')).not.toBeInTheDocument()

    // Tags section
    expect(queryByCy('side-nav-all-tags')).not.toBeInTheDocument()

    // Placeholder for when we might add more to the account sidebar
    expect(queryByCy('side-nav-account')).toBeInTheDocument()

    // Snapshot
    expect(queryByCy('side-nav')).toMatchSnapshot()
  })

  it('renders the Home side nav and shows topics', () => {
    const { queryByCy, queryAllByCy } = wrappedRender(<SideNav type='home' {...baseProps} />)

    // Main section
    expect(queryByCy('side-nav-home')).toBeInTheDocument()
    expect(queryByCy('side-nav-mylist')).toBeInTheDocument()
    expect(queryByCy('side-nav-discover')).toBeInTheDocument()
    expect(queryByCy('side-nav-collections')).toBeInTheDocument()

    // Filters section
    expect(queryByCy('side-nav-archive')).not.toBeInTheDocument()
    expect(queryByCy('side-nav-favorites')).not.toBeInTheDocument()
    expect(queryByCy('side-nav-highlights')).not.toBeInTheDocument()
    expect(queryByCy('side-nav-articles')).not.toBeInTheDocument()
    expect(queryByCy('side-nav-videos')).not.toBeInTheDocument()

    // Tags section
    expect(queryByCy('side-nav-all-tags')).not.toBeInTheDocument()

    // Topics section
    expect(queryAllByCy(/side-nav-topic-(.+)/)).toHaveLength(2)

    // Snapshot
    expect(queryByCy('side-nav')).toMatchSnapshot()
  })
})
