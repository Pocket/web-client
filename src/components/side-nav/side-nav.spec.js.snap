// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav renders the Account side nav 1`] = `
<div
  class="s3icf2j side-nav"
  data-cy="side-nav"
>
  <nav
    role="navigation"
  >
    <button
      class="sszr5nl  "
      data-cy="side-nav-home"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.671 1.442a2.004 2.004 0 0 1 2.658 0l8 7.09c.427.378.671.92.671 1.49v9.922a1.997 1.997 0 0 1-2 1.994h-3c-1.105 0-2-.893-2-1.994v-4.985c0-.275-.175-1.072-.689-1.789-.48-.67-1.201-1.203-2.311-1.203-1.11 0-1.831.533-2.311 1.203C9.175 13.887 9 14.684 9 14.96v4.986a1.996 1.996 0 0 1-2 1.993H4c-1.105 0-2-.893-2-1.994v-9.922c0-.57.244-1.112.671-1.49l8-7.09ZM12 2.932l-8 7.09v9.922h3v-4.985c0-.723.325-1.92 1.061-2.948.77-1.074 2.049-2.038 3.939-2.038 1.89 0 3.169.963 3.939 2.038.736 1.028 1.061 2.225 1.061 2.948v4.985h3v-9.922l-8-7.09ZM7 19.944Zm.002 0H7h.002Zm0 0Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:home
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-saves"
    >
      <span
        class="i1qqph0t icon side-nav-icon inactive"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2 5a2 2 0 0 1 2-2h16a2 2 0 0 1 2 2H2Zm2 0H2v6c0 5.523 4.477 10 10 10s10-4.477 10-10V5h-2v6a8 8 0 1 1-16 0V5Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M7.293 9.293a1 1 0 0 1 1.414 0L12 12.586l3.293-3.293a1 1 0 1 1 1.414 1.414l-4 4a1 1 0 0 1-1.414 0l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="i1qqph0t icon side-nav-icon active"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 21c5.523 0 10-4.477 10-10V5a2 2 0 0 0-2-2H4c-1.105 0-2 .893-2 1.998V11c0 5.523 4.477 10 10 10ZM8.707 9.293a1 1 0 0 0-1.414 1.414l4 4a1 1 0 0 0 1.414 0l4-4a1 1 0 0 0-1.414-1.414L12 12.586 8.707 9.293Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:saves
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-discover"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.422 6.122a1 1 0 0 0-1.3 1.3l2.828 7.07a1 1 0 0 0 .557.558l7.071 2.828a1 1 0 0 0 1.3-1.3l-2.828-7.07a1 1 0 0 0-.557-.558L7.422 6.122Zm3.226 7.23 4.507 1.803-1.803-4.507-2.704 2.704Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M12 21a9 9 0 1 0 0-18 9 9 0 0 0 0 18Zm0 2c6.075 0 11-4.925 11-11S18.075 1 12 1 1 5.925 1 12s4.925 11 11 11Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:discover
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-collections"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.052 2.13h-2.59c-.921 0-1.667.896-1.667 2v12c0 .204.025.401.073.587l-.023.09c-.32 1.269.215 2.579 1.239 3.298-1.663-.225-2.956-1.919-2.956-3.975v-12c0-2.209 1.492-4 3.333-4h8.334c1.606 0 2.948 1.365 3.263 3.18l-2.087-.596c-.3-.36-.717-.584-1.176-.584h-.865l-2.146-.614a2.888 2.888 0 0 0-2.732.614z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="m9.94 19.75-2.934.786c-.889.239-1.841-.433-2.127-1.5L1.773 7.445c-.286-1.067.203-2.125 1.092-2.363l2.263-.607V3.13c0-.271.054-.53.152-.766l-2.932.786C.57 3.626-.41 5.742.163 7.876L3.27 19.467c.572 2.134 2.477 3.478 4.255 3.001l6.278-1.682L9.94 19.75zm2.115-18.703a2.893 2.893 0 0 0-1.658-.054l-.512.137h1.994c.057-.03.116-.057.176-.083z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="m20.386 5.382-7.66-2.444c-.847-.27-1.779.388-2.083 1.47L7.342 16.17c-.304 1.083.136 2.179.982 2.449l7.66 2.444c.846.27 1.778-.388 2.082-1.47L21.368 7.83c.303-1.083-.136-2.179-.982-2.449zM13.276.978c-1.692-.54-3.557.777-4.165 2.942L5.81 15.68c-.608 2.165.271 4.358 1.963 4.898l7.66 2.444c1.693.54 3.557-.777 4.165-2.942L22.9 8.32c.607-2.165-.272-4.358-1.964-4.898L13.276.978z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M11.625 6.858c.152-.541.618-.87 1.041-.736l6.128 1.956c.423.135.643.683.491 1.224-.152.541-.618.87-1.04.736l-6.129-1.956c-.423-.135-.643-.683-.49-1.224zm-1.1 3.92c.152-.541.618-.87 1.04-.736l6.129 1.956c.423.135.643.683.49 1.224-.151.541-.617.87-1.04.736l-6.128-1.956c-.424-.135-.643-.683-.491-1.224zm-1.101 3.92c.152-.541.618-.87 1.041-.736l3.064.978c.423.135.643.683.491 1.225-.152.54-.618.87-1.04.735l-3.065-.978c-.423-.135-.643-.683-.49-1.224z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:collections
    </button>
    <div
      class="sv813dg"
      data-cy="side-nav-account"
    />
  </nav>
  <div
    class="bottom-nav"
  >
    <button
      aria-label="nav:return-to-top"
      class="visible"
      data-cy="side-nav-scroll-to-top"
    >
      <span
        class="i1qqph0t icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M19.707 14.707a1 1 0 0 1-1.414 0L12 8.414l-6.293 6.293a1 1 0 0 1-1.414-1.414l7-7a1 1 0 0 1 1.414 0l7 7a1 1 0 0 1 0 1.414Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`SideNav renders the Saves side nav and shows tags 1`] = `
<div
  class="s3icf2j side-nav"
  data-cy="side-nav"
>
  <nav
    role="navigation"
  >
    <button
      class="sszr5nl  "
      data-cy="side-nav-home"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.671 1.442a2.004 2.004 0 0 1 2.658 0l8 7.09c.427.378.671.92.671 1.49v9.922a1.997 1.997 0 0 1-2 1.994h-3c-1.105 0-2-.893-2-1.994v-4.985c0-.275-.175-1.072-.689-1.789-.48-.67-1.201-1.203-2.311-1.203-1.11 0-1.831.533-2.311 1.203C9.175 13.887 9 14.684 9 14.96v4.986a1.996 1.996 0 0 1-2 1.993H4c-1.105 0-2-.893-2-1.994v-9.922c0-.57.244-1.112.671-1.49l8-7.09ZM12 2.932l-8 7.09v9.922h3v-4.985c0-.723.325-1.92 1.061-2.948.77-1.074 2.049-2.038 3.939-2.038 1.89 0 3.169.963 3.939 2.038.736 1.028 1.061 2.225 1.061 2.948v4.985h3v-9.922l-8-7.09ZM7 19.944Zm.002 0H7h.002Zm0 0Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:home
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-saves"
    >
      <span
        class="i1qqph0t icon side-nav-icon inactive"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2 5a2 2 0 0 1 2-2h16a2 2 0 0 1 2 2H2Zm2 0H2v6c0 5.523 4.477 10 10 10s10-4.477 10-10V5h-2v6a8 8 0 1 1-16 0V5Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M7.293 9.293a1 1 0 0 1 1.414 0L12 12.586l3.293-3.293a1 1 0 1 1 1.414 1.414l-4 4a1 1 0 0 1-1.414 0l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="i1qqph0t icon side-nav-icon active"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 21c5.523 0 10-4.477 10-10V5a2 2 0 0 0-2-2H4c-1.105 0-2 .893-2 1.998V11c0 5.523 4.477 10 10 10ZM8.707 9.293a1 1 0 0 0-1.414 1.414l4 4a1 1 0 0 0 1.414 0l4-4a1 1 0 0 0-1.414-1.414L12 12.586 8.707 9.293Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:saves
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-discover"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.422 6.122a1 1 0 0 0-1.3 1.3l2.828 7.07a1 1 0 0 0 .557.558l7.071 2.828a1 1 0 0 0 1.3-1.3l-2.828-7.07a1 1 0 0 0-.557-.558L7.422 6.122Zm3.226 7.23 4.507 1.803-1.803-4.507-2.704 2.704Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M12 21a9 9 0 1 0 0-18 9 9 0 0 0 0 18Zm0 2c6.075 0 11-4.925 11-11S18.075 1 12 1 1 5.925 1 12s4.925 11 11 11Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:discover
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-collections"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.052 2.13h-2.59c-.921 0-1.667.896-1.667 2v12c0 .204.025.401.073.587l-.023.09c-.32 1.269.215 2.579 1.239 3.298-1.663-.225-2.956-1.919-2.956-3.975v-12c0-2.209 1.492-4 3.333-4h8.334c1.606 0 2.948 1.365 3.263 3.18l-2.087-.596c-.3-.36-.717-.584-1.176-.584h-.865l-2.146-.614a2.888 2.888 0 0 0-2.732.614z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="m9.94 19.75-2.934.786c-.889.239-1.841-.433-2.127-1.5L1.773 7.445c-.286-1.067.203-2.125 1.092-2.363l2.263-.607V3.13c0-.271.054-.53.152-.766l-2.932.786C.57 3.626-.41 5.742.163 7.876L3.27 19.467c.572 2.134 2.477 3.478 4.255 3.001l6.278-1.682L9.94 19.75zm2.115-18.703a2.893 2.893 0 0 0-1.658-.054l-.512.137h1.994c.057-.03.116-.057.176-.083z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="m20.386 5.382-7.66-2.444c-.847-.27-1.779.388-2.083 1.47L7.342 16.17c-.304 1.083.136 2.179.982 2.449l7.66 2.444c.846.27 1.778-.388 2.082-1.47L21.368 7.83c.303-1.083-.136-2.179-.982-2.449zM13.276.978c-1.692-.54-3.557.777-4.165 2.942L5.81 15.68c-.608 2.165.271 4.358 1.963 4.898l7.66 2.444c1.693.54 3.557-.777 4.165-2.942L22.9 8.32c.607-2.165-.272-4.358-1.964-4.898L13.276.978z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M11.625 6.858c.152-.541.618-.87 1.041-.736l6.128 1.956c.423.135.643.683.491 1.224-.152.541-.618.87-1.04.736l-6.129-1.956c-.423-.135-.643-.683-.49-1.224zm-1.1 3.92c.152-.541.618-.87 1.04-.736l6.129 1.956c.423.135.643.683.49 1.224-.151.541-.617.87-1.04.736l-6.128-1.956c-.424-.135-.643-.683-.491-1.224zm-1.101 3.92c.152-.541.618-.87 1.041-.736l3.064.978c.423.135.643.683.491 1.225-.152.54-.618.87-1.04.735l-3.065-.978c-.423-.135-.643-.683-.49-1.224z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:collections
    </button>
    <h4
      class="sv813dg"
    >
      nav:filters
    </h4>
    <button
      class="sszr5nl  "
      data-cy="side-nav-archive"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M1 4a2 2 0 0 1 2-2h18a2 2 0 0 1 2 2v2a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V4Zm2 0v2h18V4H3Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M3 8a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2v10a4 4 0 0 1-4 4H7a4 4 0 0 1-4-4V8Zm2 10a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V8H5v10Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M15.707 11.293a1 1 0 0 1 0 1.414l-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 1 1 1.414-1.414L11 14.586l3.293-3.293a1 1 0 0 1 1.414 0Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:archive
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-favorites"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 1a1 1 0 0 1 .897.557l2.706 5.484 6.051.88a1 1 0 0 1 .555 1.705l-4.38 4.268 1.034 6.027a1 1 0 0 1-1.45 1.054L12 18.13l-5.413 2.845a1 1 0 0 1-1.45-1.054l1.033-6.027-4.379-4.268a1 1 0 0 1 .555-1.706l6.051-.88 2.706-5.483A1 1 0 0 1 12 1Zm0 3.26L9.958 8.397a1 1 0 0 1-.753.548l-4.567.663 3.305 3.221a1 1 0 0 1 .287.885l-.78 4.548 4.085-2.147a1 1 0 0 1 .93 0l4.085 2.147-.78-4.548a1 1 0 0 1 .287-.885l3.305-3.22-4.567-.664a1 1 0 0 1-.753-.548L12 4.26Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:favorites
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-highlights"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5.512 8.546a3 3 0 0 0-.51 4.195l-.724.723a3.001 3.001 0 0 0-.586 3.415L1.18 19.391a1 1 0 0 0 .39 1.656l4.243 1.414a1 1 0 0 0 1.023-.241l1.098-1.098a3.001 3.001 0 0 0 3.415-.586l.746-.746a3.001 3.001 0 0 0 4.133-.77l6.406-9.15a3 3 0 0 0-.337-3.842l-4.112-4.112a3 3 0 0 0-3.98-.233L5.512 8.546Zm9.932-5.294-8.693 6.863a1 1 0 0 0-.087 1.492l6.4 6.4a1 1 0 0 0 1.526-.134l6.405-9.15a1 1 0 0 0-.112-1.28L16.771 3.33a1 1 0 0 0-1.327-.078ZM6.4 14.172l-.707.707a1 1 0 0 0 0 1.414l2.829 2.828a1 1 0 0 0 1.414 0l.707-.707L6.4 14.172Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:highlights
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-articles"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M17 4H7a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2ZM7 2a4 4 0 0 0-4 4v12a4 4 0 0 0 4 4h10a4 4 0 0 0 4-4V6a4 4 0 0 0-4-4H7Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M7 8a1 1 0 0 1 1-1h8a1 1 0 1 1 0 2H8a1 1 0 0 1-1-1ZM7 12a1 1 0 0 1 1-1h8a1 1 0 1 1 0 2H8a1 1 0 0 1-1-1ZM7 16a1 1 0 0 1 1-1h4a1 1 0 1 1 0 2H8a1 1 0 0 1-1-1Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
       
      nav:articles
    </button>
    <button
      class="sszr5nl  "
      data-cy="side-nav-videos"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="18"
            rx="3"
            stroke="currentColor"
            stroke-width="2"
            width="18"
            x="3"
            y="3"
          />
          <path
            d="M9 14.461V9.54a1 1 0 0 1 1.406-.914l5.538 2.461c.792.352.792 1.476 0 1.828l-5.538 2.461A1 1 0 0 1 9 14.461z"
            fill="currentColor"
          />
        </svg>
      </span>
       
      nav:videos
    </button>
    <h4
      class="sv813dg"
    >
      nav:tags
    </h4>
    <button
      class="sszr5nl  "
      data-cy="side-nav-all-tags"
    >
      <span
        class="i1qqph0t icon side-nav-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2 4a2 2 0 0 1 2-2h6a2 2 0 0 1 1.414.586L20 11.172a4 4 0 0 1 0 5.656L16.828 20a4 4 0 0 1-5.656 0l-8.586-8.586A2 2 0 0 1 2 10V4Zm8 0 8.586 8.586a2 2 0 0 1 0 2.828l-3.172 3.172a2 2 0 0 1-2.828 0L4 10V4h6Z"
            fill-rule="evenodd"
          />
          <path
            d="M9 7.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Z"
          />
        </svg>
      </span>
       
      nav:all-tags
    </button>
    <button
      class="sszr5nl  tag-class"
      data-cy="side-nav-tags-home"
    >
      home
    </button>
    <button
      class="sszr5nl  tag-class"
      data-cy="side-nav-tags-space"
    >
      space
    </button>
  </nav>
  <div
    class="bottom-nav"
  >
    <button
      aria-label="nav:return-to-top"
      class="visible"
      data-cy="side-nav-scroll-to-top"
    >
      <span
        class="i1qqph0t icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M19.707 14.707a1 1 0 0 1-1.414 0L12 8.414l-6.293 6.293a1 1 0 0 1-1.414-1.414l7-7a1 1 0 0 1 1.414 0l7 7a1 1 0 0 1 0 1.414Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
